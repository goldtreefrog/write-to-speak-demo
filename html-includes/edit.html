<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="keywords" content="">
  <meta name="description" content="">
  <meta name="robots" content="">
  <meta name="author" content="Margaret Blauvelt">
  <title>Write to Speak</title>
  <link rel="stylesheet" href="index.css">
</head>

<body>
  <main id="write" role="main">
    <section id="snippets">
      <form action="#">
        <h2>Edit Snippets</h2>
        <h3>Click on the snippet you wish to change</h3> @@include('./partials/_snippets.html') @@include('./partials/_edit-box-label.html') @@include('./partials/_write-box.html')
        <button type="submit" name="button" id="update-snippet">Update</button>
      </form>
    </section>
  </main>

</body>

</html>