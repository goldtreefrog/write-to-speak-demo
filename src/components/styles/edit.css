section#edit {
  width: 98%;
}
section#edit p.page-instructions {
  text-align: left;
  margin: 16px 2% 8px;
}
.snippet-line {
  display: block;
  position: relative;
  width: 98.75%;
  margin: 0 auto 12px;
  overflow: hidden;
  text-overflow: ellipsis;
  background-color: #81f986;
  height: auto;
  border-radius: 5px;
  border: solid 2px #111;
  padding-bottom: 3px;
}

.snippet-edit {
  background-color: #81f986;
  width: 100%;
  margin-bottom: -6px;
  border: none;
}
div.snippet-edit:after {
  background-color: #81f986;
}
.snippet-line .snippet-buttons-container {
  display: inline-block;
  width: auto;
  padding: 0;
}

button.snippet-button-talk,
button.snippet-button-edit,
button.snippet-delete {
  display: inline-block;
  position: relative;
  margin-left: 2px;
  margin-right: 2px;
  width: 4em;
  height: 1.5em;
  color: black;
  padding: 1px 5px;
  border-radius: 8px;
  font-size: 0.9rem;
}

section#edit .disabled,
div.snippet-line .disabled:after,
div.disabled,
div.disabled:hover,
div.disabled:after {
  color: #666;
  background: none;
  background-color: #ccc;
  background: linear-gradient(#ccc, #ccc);
  border-color: #aaa;
}
/* Make the ellipsis background change background color with everything else */
div.snippet:hover:after,
div.snippet:focus:after {
  background: inherit;
}

/* Tablet or small screen, not phone, 2 snippets across */
@media only screen and (min-width: 472px) {
  /* NOTE: .snippet-line is similar to .page-snippets found in talk.css */
  .snippet-line {
    margin: 0 1% 12px;
    width: 48%;
    display: inline-block;
    position: relative;
    float: left;
    clear: none;
  }
  .snippet-edit {
    width: 100%;
    height: 3.5em;
    float: left;
  }
}

@media only screen and (min-width: 738px) {
  .snippet-line {
    width: 31.3%;
  }
}

/* Non-mobile, snippets  4 snippets across */
@media only screen and (min-width: 1000px) {
  .snippet-line {
    width: 23%;
  }
  .snippet-edit {
  }
}
