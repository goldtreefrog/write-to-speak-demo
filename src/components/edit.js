import React, { Component } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import Aux from "./../hoc/_aux.js";
import Snippet from "./snippet";
import SnippetDelete from "./snippet-delete";
import WritingArea from "./writing-area";
// import SpellingArea from "./spelling-area";
import Feedback from "./feedback";
import SayIt from "./say-it";
import "./styles/edit.css";
import {
  writingAreaPopulate,
  writingAreaReset,
  setSnippetsAvailability,
  clearFeedback,
  giveFeedback,
  setWhatToSay,
  clearWhatToSay,
  deleteSnippet
} from "./../store/actions";

// Use named export for unconnected component (for tests)
export class Edit extends Component {
  componentWillUnmount = () => {
    this.props.dispatch(writingAreaReset());
    this.props.dispatch(setSnippetsAvailability({ snippetsAvail: true }));
  };

  loadSnippetForUpdate = e => {
    // Remove any previous feedback and text to speak
    this.props.dispatch(clearFeedback());
    this.props.dispatch(clearWhatToSay());

    // Copy snippet text into writing textarea.
    this.props.dispatch(
      writingAreaPopulate({
        activeSnippetId: e.target.id,
        activeSnippetText: e.target.value
      })
    );

    // Make all snippets disabled and grayed out, except for the one we are updating, which we give a different color scheme.
    this.props.dispatch(setSnippetsAvailability({ snippetsAvail: false }));

    window.scrollTo(0, 0);
  };

  deleteSnippetClicked = e => {
    e.preventDefault();

    this.props.dispatch(
      deleteSnippet({
        userId: this.props.currentUser._id,
        snippetId: e.target.value
      })
    );

    let feedbackMsg = "Snippet deleted.";
    this.props.dispatch(giveFeedback({ feedback: feedbackMsg }));
    this.props.dispatch(
      setWhatToSay({
        whatToSay: feedbackMsg,
        useVoice: "UK English Female"
      })
    );
  };

  render() {
    if (!this.props.loggedIn) {
      localStorage.setItem("showFeedbackFlag", "t");
      let feedbackMsg = "Please sign in.";
      this.props.dispatch(giveFeedback({ feedback: feedbackMsg }));
      this.props.dispatch(
        setWhatToSay({
          whatToSay: feedbackMsg,
          useVoice: "UK English Female"
        })
      );
      return <Redirect to="/login" />;
    }

    let snippetInstructions = () => {
      if (this.props.snippets.snippetsAvail) {
        return "Click on the snippet you wish to change. To delete it, click the 'Delete' button.";
      }
      return "The snippets below will become active again when you finish your update";
    };

    return (
      <Aux>
        <h2>Edit</h2>
        <Feedback />
        {this.props.writing.visible && (
          <div>
            <WritingArea
              activeSnippetId={this.props.writing.activeSnippetId}
              activeSnippetText={this.props.writing.activeSnippetText}
              visible={this.props.writing.visible}
              buttonText={{
                saveUpdate: "Update Snippet",
                resetCancel: "Cancel Update"
              }}
            />
            {/* <SpellingArea spellData={this.props.spellingArea} visible={this.props.spellingArea.visible} /> */}
          </div>
        )}
        <section id="edit">
          <p>
            {snippetInstructions()}
            {this.props.snippets.snippetsAvail && <SayIt />}
          </p>
          {this.props.snippets.snippets.map(snippet => (
            <div className="snippet-line" key={"div" + snippet._id}>
              <Snippet
                className="snippet snippet-edit"
                click={() => this.loadSnippetForUpdate}
                text={snippet.snippetText}
                id={snippet._id}
                orderkey={snippet.orderkey}
                // key={snippet._id}
                disabled={!this.props.snippets.snippetsAvail}
              />
              <SnippetDelete
                click={() => this.deleteSnippetClicked}
                id={snippet._id}
                ishidden={!this.props.snippets.snippetsAvail}
              />
            </div>
          ))}
        </section>
      </Aux>
    );
  }
}

const mapStateToProps = state => {
  return {
    snippets: state.snippets,
    writing: state.writing,
    // spellingArea: state.spelling.spellingArea,
    other: state.other,
    currentUser: state.auth.currentUser,
    loggedIn: state.auth.currentUser !== null
  };
};

// Use default export for the connected component (for app)
export default connect(mapStateToProps)(Edit);
