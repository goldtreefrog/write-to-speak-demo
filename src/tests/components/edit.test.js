import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { Edit } from "./../../components/edit.js";
import { spellDataInitial as spellData, activeSnippetId } from "./../fixtures/spell-data.js";
import { snippets } from "./../fixtures/snippet-data.js";

it("renders without crashing and matches snapshot", () => {
  const wrapper = shallow(<Edit writing={{ visible: true }} spellingArea={spellData} snippets={snippets} />);
  console.log(wrapper.debug);
  expect(wrapper).toMatchSnapshot();
});
