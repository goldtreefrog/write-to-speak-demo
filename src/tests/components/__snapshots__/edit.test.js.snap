// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<Aux>
  <h2>
    Edit
  </h2>
  <Connect(Feedback) />
  <div>
    <Connect(WritingArea)
      buttonText={
        Object {
          "resetCancel": "Cancel Update",
          "saveUpdate": "Update Snippet",
        }
      }
      visible={true}
    />
  </div>
  <section
    id="edit"
  >
    <p>
      The snippets below will become active again when you finish your update
    </p>
    <Snippet
      click={[Function]}
      disabled={true}
      id={123}
      key="123"
      orderkey={0}
      text="Hi there"
    />
    <Snippet
      click={[Function]}
      disabled={true}
      id={312}
      key="312"
      orderkey={3}
      text="Mom"
    />
    <Snippet
      click={[Function]}
      disabled={true}
      id={666}
      key="666"
      orderkey={2}
      text="Dad"
    />
    <Snippet
      click={[Function]}
      disabled={true}
      id={427}
      key="427"
      orderkey={1}
      text="come talk with me"
    />
    <Snippet
      click={[Function]}
      disabled={true}
      id={555}
      key="555"
      orderkey={6}
      text="Once upon a time, when it was a dark and stormy night, Sandy heard a tremendously loud 'boom!'"
    />
  </section>
</Aux>
`;
